File: kotlin/06-io-operations/SafeRead.kt
import java.io.File
import java.io.FileNotFoundException

fun main() {
    val file = File("missing.txt")
    try {
        val text = file.readText()
        println(text)
    } catch (e: FileNotFoundException) {
        println("File not found: ${file.name}")
    }
}

File: typescript/06-io-operations/safe_read.ts
import { readFile } from "node:fs/promises";

async function main(): Promise<void> {
    try {
        const data: string = await readFile("missing.txt", "utf-8");
        console.log(data);
    } catch (err: unknown) {
        // Narrow from unknown before accessing any properties
        if (err instanceof Error && "code" in err && err.code === "ENOENT") {
            console.log("File not found: missing.txt");
        } else {
            throw err;
        }
    }
}

main();
